Java 8 Maven Projects fail with 2.0.1 Build Snapshot (invalid flag: --release)

### Description

A new project created with 2.0.1 Build Snapshot for Java 8 with Maven will not build.  I observed this when debugging failing GitHub Actions for #319 

### Task List

- [x] Steps to reproduce provided
- [x] Stacktrace (if present) provided
- [x] Example that reproduces the problem uploaded to Github
- [x] Full description of the issue provided (see below)

### Steps to Reproduce

1. Create a new app with Java 8, and Maven [Here's one](https://github.com/JasonTypesCodes/mn2-mvn-jdk8)
2. Using JDK 8, try to build the application

### Expected Behaviour

The application should build

### Actual Behaviour

The application does not build:
```
[INFO] Scanning for projects...
[INFO] 
[INFO] --------------------------< com.example:demo >--------------------------
[INFO] Building demo 0.1
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-resources-plugin:3.1.0:resources (default-resources) @ demo ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 2 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile) @ demo ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 1 source file to /home/schindlerj/src/temp/mn2-mvn-jdk8/target/classes
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 0.941 s
[INFO] Finished at: 2020-07-08T23:21:18-05:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project demo: Fatal error compiling: invalid flag: --release -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException

```

### Environment Information

- **Operating System**: Fedora Linux Workstation 31
- **Micronaut Version:** 2.0.1.BUILD-SNAPSHOT
- **JDK Version:** OpenJDK Runtime Environment (build 1.8.0_252-b09)

### Example Application

https://github.com/JasonTypesCodes/mn2-mvn-jdk8

My earlier comment is only partially correct.  Unsetting `maven.compiler.release` fixes `compile` but `testCompile` still fails with
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:testCompile (default-testCompile) on project something: Execution default-testCompile of goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:testCompile failed: For input string: ""
```

This is because of a call to `Integer.valueOf` here: https://github.com/apache/maven-compiler-plugin/blob/maven-compiler-plugin-3.8.1/src/main/java/org/apache/maven/plugin/compiler/TestCompilerMojo.java#L305

An issue (https://issues.apache.org/jira/browse/MCOMPILER-339) exists in the Maven Compiler Plugin Jira requesting that Maven just ignore release when the JDK being used is < 9.

A [comment](https://issues.apache.org/jira/browse/MCOMPILER-339?focusedCommentId=17068524&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-17068524) there suggests:

> It is extremely  dangerous to ignore the release flag. What should work for you is  a profile activated by JDK:
```
    <profile>
      <id>javac-release</id>
      <activation>
        <jdk>[11,)</jdk>
      </activation>
      <properties>
        <maven.compiler.release>8</maven.compiler.release>
      </properties>
    </profile>
```

--- a/starter-core/src/main/java/io/micronaut/starter/feature/Features.java
+++ b/starter-core/src/main/java/io/micronaut/starter/feature/Features.java
@@ -80,6 +80,10 @@ public class Features extends ArrayList<String> {
         return featureList;
     }
 
+    public JdkVersion javaVersion() {
+        return javaVersion;
+    }
+
     public String getTargetJdk() {
         if (language().isJava() && testFramework().isJunit()) {
             return VersionInfo.toJdkVersion(javaVersion.majorVersion());

--- a/starter-core/src/main/java/io/micronaut/starter/feature/build/maven/templates/pom.rocker.raw
+++ b/starter-core/src/main/java/io/micronaut/starter/feature/build/maven/templates/pom.rocker.raw
@@ -5,6 +5,7 @@
 @import io.micronaut.starter.feature.build.maven.templates.dependency
 @import io.micronaut.starter.build.Property
 @import io.micronaut.starter.application.ApplicationType
+@import io.micronaut.starter.options.JdkVersion
 
 @args (
 ApplicationType applicationType,
@@ -29,6 +30,14 @@ List<Property> properties
 
   <properties>
     <jdk.version>@features.getTargetJdk()</jdk.version>
+@if (features.javaVersion() == JdkVersion.JDK_8) {
+    <!-- If you are building with JDK 9 or higher, you can uncomment the lines below to set the release version -->
+    <!-- <release.version>@features.javaVersion().majorVersion()</release.version> -->
+    <!-- <maven.compiler.release>${release.version}</maven.compiler.release> -->
+} else {
+    <release.version>@features.javaVersion().majorVersion()</release.version>
+    <maven.compiler.release>${release.version}</maven.compiler.release>
+}
     <maven.compiler.target>${jdk.version}</maven.compiler.target>
     <maven.compiler.source>${jdk.version}</maven.compiler.source>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>

